browser/ai_chat/android/ai_chat_utils_android.cc

```diff
@@ -51,5 +53,33 @@ static void JNI_BraveLeoUtils_OpenLeoQuery(
       base::android::ConvertJavaStringToUTF8(query), std::nullopt, std::nullopt,
       base::Time::Now(), std::nullopt, false);
   conversation->SubmitHumanConversationEntry(std::move(turn));
+
+  content::OpenURLParams params(
+      GURL(base::StrCat({kChatUIURL, conversation->get_conversation_uuid()})),
+      content::Referrer(), WindowOpenDisposition::CURRENT_TAB,
+      ui::PAGE_TRANSITION_FROM_API, false);
+  CHECK(web_contents->OpenURL(params, {}));
+}
+
+static void JNI_BraveLeoUtils_OpenLeoUrlForTab(
+    JNIEnv* env,
+    const base::android::JavaParamRef<jobject>& jweb_contents) {
+  content::WebContents* web_contents =
+      content::WebContents::FromJavaWebContents(jweb_contents);
+  AIChatService* ai_chat_service = AIChatServiceFactory::GetForBrowserContext(
+      web_contents->GetBrowserContext());
+  DCHECK(ai_chat_service);
```

Comment:
Aren't we meant to not use DCHECK anymore and either CHECK or gracefully handle?